spec/hyacinth/dynamic_fields_map_spec.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Hyacinth::DynamicFieldsMap do
+  describe '.generate' do
+    context 'when there aren\'t any categories present' do
+      subject(:map) { described_class.generate('item_rights') }
+
+      it 'returns empty map' do
+        expect(map).to eql({})
+      end
+    end
+
+    context 'when generating map for asset_rights form' do
+      subject(:map) { described_class.generate('asset_rights') }
+      before { Hyacinth::DynamicFieldsLoader.load_rights_fields! }
```

Comment:
This `before` step definitely does test the `Hyacinth::DynamicFieldsMap`'s `generate` method, but I think it might be better to write a test with mock data so that we don't need to update this test if/when the hard-coded set of rights fields change.  Swapping it with `load_fields!(...)` and some mock json should do it, and would better isolate the success of this test from the success of the `load_rights_fields!` implementation.

Reply:
Done.